batch/batches/Delete/Engines/KDeletingEntryEngine.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * @package Scheduler
+ * @subpackage Delete
+ */
+class KDeletingEntryEngine extends KDeletingEngine
+{
+	/* (non-PHPdoc)
+	 * @see KDeletingEngine::delete()
+	 */
+	protected function delete(KalturaFilter $filter)
+	{
+		return $this->deleteEntries($filter);
+	}
+	
+	/**
+	 * @param KalturaBaseEntryFilter $filter The filter should return the list of entries that need to be deleted
+	 * @return int the number of deleted entries
+	 */
+	protected function deleteEntries(KalturaBaseEntryFilter $filter)
+	{
+		if (!$filter->orderBy)
+		{
+			$filter->orderBy = KalturaBaseEntryOrderBy::UPDATED_AT_ASC;
+		}
+		
+		$entriesList = KBatchBase::$kClient->baseEntry->listAction($filter, $this->pager);
+		if (!$entriesList->objects || !count($entriesList->objects))
+		{
+			return 0;
+		}
+		
+		$entriesToDeletePerRequest = KBatchBase::$taskConfig->params->entriesToDeletePerRequest;
+		$waitBetweenRequestsInSeconds = KBatchBase::$taskConfig->params->waitBetweenRequestsInSeconds;
+		
+		$entriesCount = 0;
```

Comment:
worth changing to something more meaningful like handled or deleted entries count
also whats the difference between countResults and entriesCount? looks like they should be similar at the end